Bug: `waypoint runner agent -remote` panics

<!-- Please reserve GitHub issues for bug reports and feature requests.

For questions, the best place to get answers is on our [discussion forum](https://discuss.hashicorp.com/c/waypoint), as they will get more visibility from experienced users than the issue tracker.

Please note: We take Waypoint's security and our users' trust very seriously. If you believe you have found a security issue in Waypoint, please responsibly disclose by contacting us at security@hashicorp.com. Our PGP key is available at our security page: https://www.hashicorp.com/security/

-->

**Describe the bug**
Running `waypoint runner agent` with the `-remote` flag causes a panic for me under all circumstances.

Trace:
```
$ waypoint runner agent -remote

panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x10 pc=0x319bb61]

goroutine 1 [running]:
github.com/hashicorp/waypoint/internal/cli.(*baseCommand).Init(0xc000c00000, 0xc000d3fcb0, 0x4, 0x4, 0xc000caa3f0, 0xc000d3fb28)
	/home/circleci/project/waypoint/internal/cli/base.go:253 +0x821
github.com/hashicorp/waypoint/internal/cli.(*RunnerAgentCommand).Run(0xc000792c40, 0xc00040dab0, 0x1, 0x1, 0x0)
	/home/circleci/project/waypoint/internal/cli/runner_agent.go:40 +0x218
github.com/mitchellh/cli.(*CLI).Run(0xc000ca4000, 0xc000d3fee8, 0x3e2bf18, 0xc0003e84d0)
	/home/circleci/go/pkg/mod/github.com/mitchellh/cli@v1.1.2/cli.go:262 +0x41a
github.com/hashicorp/waypoint/internal/cli.Main(0xc00040da80, 0x4, 0x4, 0x0)
	/home/circleci/project/waypoint/internal/cli/main.go:119 +0x57a
main.main()
	/home/circleci/project/waypoint/cmd/waypoint/main.go:14 +0xa5
```

**Expected behavior**

I am honestly not sure what this flag is supposed to be doing. I ran across it accidentally while trying to set up a standalone runner.

**Waypoint Platform Versions**
Additional version and platform information to help triage the issue if
applicable:

* Waypoint CLI Version: 0.4.0
* Waypoint Server Platform and Version: 0.4.0
* Waypoint Plugin: any

**Additional context**

It's crashing [here](https://github.com/hashicorp/waypoint/blob/321eb68d56e918968b0145392158826a9b77e5e8/internal/cli/base.go#L253), specifically when inspecting `c.cfg.Runner`.  The crazy thing is that c.cfg is, in fact, not nil _until_ you ask for `c.cfg.Runner`

It seems like a funky interaction with the embedded [*hclConfig](https://github.com/hashicorp/waypoint/blob/321eb68d56e918968b0145392158826a9b77e5e8/internal/config/config.go#L16), which is nil in this case?

https://github.com/hashicorp/waypoint/blob/321eb68d56e918968b0145392158826a9b77e5e8/internal/config/config.go#L15-L31

I also tried running it in a project directory with an hcl config that had a runner stanza, which I was kind of hoping would work based on the [error message](https://github.com/hashicorp/waypoint/blob/321eb68d56e918968b0145392158826a9b77e5e8/internal/cli/base.go#L255-L259) I think this panic is masking.

--- a/.changelog/1708.txt
+++ b/.changelog/1708.txt
@@ -1,4 +1,4 @@
 ```release-note:bug
-core/runner: prevent use of operation flags on `runner agent` command
+cli: prevent use of operation flags on `runner agent` command
 ```
 

--- a/.changelog/1708.txt
+++ b/.changelog/1708.txt
@@ -1,0 +1,4 @@
+```release-note:bug
+core/runner: prevent use of operation flags on `runner agent` command
+```
+

--- a/internal/cli/runner_agent.go
+++ b/internal/cli/runner_agent.go
@@ -191,7 +191,7 @@ func (c *RunnerAgentCommand) Run(args []string) int {
 }
 
 func (c *RunnerAgentCommand) Flags() *flag.Sets {
-	return c.flagSet(flagSetOperation, func(set *flag.Sets) {
+	return c.flagSet(0, func(set *flag.Sets) {
 		f := set.NewSet("Command Options")
 		f.BoolVar(&flag.BoolVar{
 			Name:   "enable-dynamic-config",

--- a/website/content/commands/runner-agent.mdx
+++ b/website/content/commands/runner-agent.mdx
@@ -23,13 +23,6 @@ Usage: `waypoint runner agent [options]`
 - `-app=<string>` - App to target. Certain commands require a single app target for Waypoint configurations with multiple apps. If you have a single app, then this can be ignored.
 - `-workspace=<string>` - Workspace to operate in.
 
-#### Operation Options
-
-- `-label=<key=value>` - Labels to set for this operation. Can be specified multiple times.
-- `-remote` - True to use a remote runner to execute. This defaults to false
-  unless 'runner.default' is set in your configuration.
-- `-remote-source=<key=value>` - Override configurations for how remote runners source data. This is specified to the data source type being used in your configuration. This is used for example to set a specific Git ref to run against.
-
 #### Command Options
 
 - `-enable-dynamic-config` - Allow dynamic config to be created when an exec plugin is used.

